mlflow/utils/model_utils.py

```diff
@@ -191,3 +194,70 @@ def _validate_onnx_session_options(onnx_session_options):
                 raise ValueError(
                     f"Value for key {key} in onnx_session_options should be >= 0, not {value}"
                 )
+
+
+def _update_inference_config(
+    pyfunc_config: Dict[str, Any], load_config: Dict[str, Any], logger
+) -> Dict[str, Any]:
+    """
+    Updates the inference configuration according to the model's configuration and the overrides.
+    Only arguments already present in the inference configuration can be updated. The environment
+    variable ``MLFLOW_PYFUNC_INFERENCE_CONFIG`` can also be used to provide additional inference
+    configuration.
+    """
+
+    overrides = {}
+    if env_overrides := os.getenv("MLFLOW_PYFUNC_INFERENCE_CONFIG"):
+        logger.debug(
+            "Inference configuration is being loaded from ``MLFLOW_PYFUNC_INFERENCE_CONFIG``"
+            " environ."
+        )
+        overrides.update(dict(json.loads(env_overrides)))
+
+    if load_config:
+        overrides.update(load_config)
```

Comment:
Q:

What if `MLFLOW_PYFUNC_INFERENCE_CONFIG` and load_config both exists for the same param key ?
shall we forbid this case or tell user which one will be overridden by another one ?

Reply:
Good question. In the original design doc we stablished a priority and overriding mechanism that follows the following priority:
1. Values indicated in the parameter `inference_config` of `load_model`.
2. Values indicated in the environ `MLFLOW_PYFUNC_INFERENCE_CONFIG`.
3. Values indicated in the model metadata when using `log_model` or `save_model`.

However, I see your point of this maybe being complicated. We can allow overriding from 1) to 3) while throw and exception when 1) and 2) are both indicated. We can also keep this priority and start warning when overriding is happening. What do you think?